src/librustc_passes/ast_validation.rs

```diff
@@ -167,11 +168,58 @@ impl<'a> AstValidator<'a> {
                 "only lifetime parameters can be used in this context");
         }
     }
+
+    /// With eRFC 2497, we need to check whether an expression is ambigious and warn or error
+    /// depending on the edition, this function handles that.
+    fn while_if_let_ambiguity(&self, expr: &P<Expr>) {
+        if let Some((span, op_kind)) = self.while_if_let_expr_ambiguity(&expr) {
+            let mut err = self.err_handler().struct_span_err(
+                span, &format!("ambigious use of `{}`", op_kind.to_string())
+            );
+
+            err.note(
+                "This will be a error until the `let_chains` feature is stabilized."
```

Comment:
Most other diagnostic `note`s are written with no initial capital and no period (arbitrarily-chosen examples: [1](https://github.com/rust-lang/rust/blob/28bcffead74d5e17c6cb1f7de432e37f93a6b50c/src/librustc/traits/error_reporting.rs#L1513) [2](https://github.com/rust-lang/rust/blob/28bcffead74d5e17c6cb1f7de432e37f93a6b50c/src/librustc_typeck/check/method/suggest.rs#L405) [3](https://github.com/rust-lang/rust/blob/28bcffead74d5e17c6cb1f7de432e37f93a6b50c/src/librustc_typeck/astconv.rs#L1070)). (I'm in favor of this convention because it offers visual consistency while not requiring all notes to be complete sentences.)

Reply:
Thanks, fixed!